include/inexor/vulkan-renderer/vk_tools/representation.hpp

```diff
@@ -6,13 +6,9 @@
 
 namespace inexor::vulkan_renderer::vk_tools {
 
-/// @brief Convert a Vulkan data structure into a std::string_view.
-template <typename VulkanDataType>
-[[nodiscard]] std::string_view as_string(VulkanDataType);
+template <typename T>
```

Comment:
Since we have template specialization, I added the documentation to them in the .cpp file.
Should I add something here as well?

Reply:
Docs should be always inside the header.
Also i would add all available specializations here too? Not sure about this.

Reply:
Yeah I will do that. The .cpp file will contain only the definition of the other functions then.

Reply:
I think doc comments can go in the source file? If not, we'd have to stick with a single one in the header file like we did before since the specializations need to be in the source file.

Reply:
I would never place them inside the source files as the source files may not be shipped everytime, e.g. libraries also mode ide will port you to the header if you click on the function. Also the documentation is not between so many code.

Reply:
I moved the documentation of `std::string_view result_to_description(VkResult result)` to the header file and added a comment on `std::string_view as_string(T)`. Done